docker/Dockerfile

```diff
@@ -73,7 +73,7 @@ RUN python3 -m venv $VIRTUAL_ENV
 ENV PATH="$VIRTUAL_ENV/bin:$PATH"
 RUN rm -rf tmp && git clone --depth 1 --branch ${LIGHTNING_VERSION} https://github.com/PennyLaneAI/pennylane-lightning.git tmp\
     && mv tmp/* /opt/pennylane-lightning && rm -rf tmp
-RUN pip install --no-cache-dir build cmake ninja pybind11 toml wheel
+RUN pip install --no-cache-dir build cmake ninja pybind11 toml wheel setuptools>=75.8.1
```

Comment:
Do we also need to update [setuptools pinned version](https://github.com/PennyLaneAI/pennylane-lightning/blob/98a9292f711573a8aae32978ad0e59e8ba18d261/scripts/configure_pyproject_toml.py#L94)?

Reply:
nwm -- we need to sync with compilation before changing the pinned version to keep the consistency across PL. We'll address this in a followup PR